pkg/hubble/exporter/dynamic_exporter.go

```diff
@@ -0,0 +1,158 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package exporter
+
+import (
+	"context"
+	"errors"
+
+	"github.com/sirupsen/logrus"
+
+	v1 "github.com/cilium/cilium/pkg/hubble/api/v1"
+	"github.com/cilium/cilium/pkg/hubble/exporter/exporteroption"
+	"github.com/cilium/cilium/pkg/lock"
+	"github.com/cilium/cilium/pkg/time"
+)
+
+type dynamicExporter struct {
+	FlowLogExporter
+	logger           logrus.FieldLogger
+	watcher          *configWatcher
+	managedExporters map[string]*managedExporter
+	maxFileSizeMB    int
+	maxBackups       int
+	mutex            lock.RWMutex
+}
+
+// OnDecodedEvent distributes events across all managed exporters.
+func (d *dynamicExporter) OnDecodedEvent(ctx context.Context, event *v1.Event) (bool, error) {
+	select {
+	case <-ctx.Done():
+		return false, d.Stop()
+	default:
+	}
+
+	d.mutex.RLock()
+	defer d.mutex.RUnlock()
+
+	var errs error
+	for _, me := range d.managedExporters {
+		if me.config.End == nil || me.config.End.After(time.Now()) {
+			_, err := me.exporter.OnDecodedEvent(ctx, event)
+			errs = errors.Join(errs, err)
+		}
+	}
+	return false, errs
+}
+
+// Stop stops configuration watcher  and all managed flow log exporters.
+func (d *dynamicExporter) Stop() error {
+	d.mutex.Lock()
+	defer d.mutex.Unlock()
+
+	d.watcher.Stop()
+
+	var errs error
+	for _, me := range d.managedExporters {
+		errs = errors.Join(errs, me.exporter.Stop())
+	}
+
+	return errs
+}
+
+// NewDynamicExporter creates instance of dynamic hubble flow exporter.
+func NewDynamicExporter(logger logrus.FieldLogger, configFilePath string, maxFileSizeMB, maxBackups int) *dynamicExporter {
+	dynamicExporter := &dynamicExporter{
+		logger:           logger,
+		managedExporters: make(map[string]*managedExporter),
+		maxFileSizeMB:    maxFileSizeMB,
+		maxBackups:       maxBackups,
+	}
+
+	registerMetrics(dynamicExporter)
+
+	watcher := NewConfigWatcher(configFilePath, dynamicExporter.onConfigReload)
+	dynamicExporter.watcher = watcher
+	return dynamicExporter
+}
+
+func (d *dynamicExporter) onConfigReload(ctx context.Context, hash uint64, config DynamicExportersConfig) {
+	d.mutex.Lock()
+	defer d.mutex.Unlock()
+
+	configuredFlowLogNames := make(map[string]bool)
+	for _, flowlog := range config.FlowLogs {
+		configuredFlowLogNames[flowlog.Name] = true
```

Comment:
```suggestion
		configuredFlowLogNames[flowlog.Name] = struct{}{}
```
nit: same comment as in `validateConfig`